### ISSUE TITLE
Markers returned by addMarker() are not the same that comes on the getInfoWindow() callback

### ISSUE THREAD

**Issue by riclage** [mapbox/mapbox-gl-native#5063]
**Platform:** Android API 23
**Mapbox SDK version:** com.mapbox.mapboxsdk:mapbox-android-sdk:4.0.1@aar

### Steps to trigger behavior

\1. Add a custom marker to the map:

```
Marker myMarker = map.addMarker(new MarkerOptions()
        .title("name")
        .position(pos)
        .icon(R.drawable.my_icon);
```
\2. Set a `InfoWindowAdapter`
\3. Try to compare the `Marker` returned on the callback with `myMarker`

### Expected behavior
Markers should be same object

### Actual behavior
Marker returned on `getInfoWindow()` is not the same object, not even the same id.

I tried 4.1.0-SNAPSHOT and the issue I'm reporting here seems solved in that version but I found other issues with that version of Mapbox which prevents from using it:

 - MyLocation marker color is using the "accent color" as default for its accuracy region instead of the default blue
 - Animation of MyLocation marker is very slow when location is updated
 - MyLocation marker is not placed correctly on the new updated location on camera move

Not sure how to proceed.

**Comment by tobrun** [comment]
@riclage
Thank you for reaching out and reporting your problem, this feedback is great.

> Markers returned by addMarker() are not the same that comes on the getInfoWindow() callback

I have been doing some tests with `4.0.1` and I'm unable to reproduce this issue. In my tests both id and equals method are working as designed. This looks like the issue we had with `4.0.0` but should have been resolved with `4.0.1`. Is there a bit more information to your use-case that I'm not seeing here?

> MyLocation marker color is using the "accent color" as default for its accuracy region instead of the default blue

Next release will include support for tinting UI components. To configure the color used as accuracy color you need to call `mapboxMap.getMyLocationViewSettings().setAccuracyTintColor(@ColorInt int color)`. Note that there is also a method to configure it's alpha value `setAccuracyAlpha(@IntRange(from = 0, to = 255) int alpha)`.

> Animation of MyLocation marker is very slow when location is updated

Could you add a bit more information about what you mean by slow? Is it the  duration needed to do the actual animation? or are the location updates not matching the real world location (due to entity interpolation) which could be received as been slow. 

> MyLocation marker is not placed correctly on the new updated location on camera move

I will need to look into this. I'm guessing you are updating the camera position nearby the current camera position for this?

**Comment by tobrun** [comment]
The initial issue is no longer reproducible with latests versions of the SDK. Did not receive feedback on other items mentioned in OP. Closing for now.
